apps/console/src/features/roles/components/wizard-updated/application-role-wizard.tsx

```diff
@@ -118,15 +118,19 @@ export const ApplicationRoleWizard: FunctionComponent<ApplicationRoleWizardProps
         const options: DropdownProps[] = [];
 
         subscribedAPIResourcesListData?.map((apiResource: AuthorizedAPIListItemInterface) => {
-            options.push({
-                key: apiResource?.id,
-                text: apiResource?.displayName,
-                value: apiResource?.id
-            });
+            const isNotSelected: boolean = !selectedAPIResources
+                ?.find((selectedAPIResource: APIResourceInterface) => selectedAPIResource?.id === apiResource?.id);
+
+            if (isNotSelected) {
+                options.push({
+                    key: apiResource?.id,
+                    text: apiResource?.displayName,
+                    value: apiResource?.id
+                });
+            }
         });
```

Comment:
We can simplify this as follows as well.

```suggestion
        const options: DropdownProps[] = subscribedAPIResourcesListData?.filter((apiResource: AuthorizedAPIListItemInterface) => {
            
            const isNotSelected: boolean = !selectedAPIResources
                ?.find((selectedAPIResource: APIResourceInterface) => selectedAPIResource?.id === apiResource?.id);

            return isNotSelected;
        });
```